evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -982,6 +982,17 @@ where
         // constant and the `ShiftTable`.
         let mut trace = vec![];
 
+        // Push kernel code.
```

Comment:
So this is never checked outside of recursion?
I'd like the regular verifier to be consistent with the continuations impl, as for now there is no check that the base AllStark proof is using the right kernel code.